### PR TITLE
#76 dockerize web frontend

### PR DESCRIPTION
**Pull request by edwardkcyu**
Run by:
```sh
docker-compose up web
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,15 @@
 version: '3.1'
 services:
+  web:
+    container_name: web
+    build:
+      context: .
+      dockerfile: web.Dockerfile
+    environment:
+      REACT_APP_BACKEND_URL: "http://localhost:8063"
```
Comment (edwardkcyu) [inline docker-compose.yml:9]: probably you may need to update this.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    ports:
+      - 8082:8080
```
Comment (edwardkcyu) [inline docker-compose.yml:11]: 8082 is the port for local, http://localhost:8082

File: web.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+COPY web/. .
+RUN npm run build
+
+FROM ${NODE_IMAGE}
+WORKDIR /app
+COPY --from=BUILDER /app/build .
+EXPOSE 8080
+# use http-server for simplicity, should be enough for production
+# can consider using Nginx if we hit performance issues
+ENTRYPOINT ["npx", "http-server", "."]
```
Comment (edwardkcyu) [inline web.Dockerfile:17]: As we don't need all the fancy features of Nginx, I chose the light-weight and production grade `http-server` for the quick and easy setup.

https://www.npmjs.com/package/http-server

File: docker-compose.yml (hunk 1 above)
Comment (freddy33) [reply docker-compose.yml:9]: For some reason I got `"POST /undefined/init-env" Error (404): "Not found"` which should not happen with the env set here.

Comment (edwardkcyu) [reply docker-compose.yml:9]: hmm, this should not happen, I will double check

Comment (edwardkcyu) [reply docker-compose.yml:9]: The env vars were not available during build time, I will make another PR for that.

**Pull request merged**